diff -Naur cdo-1.8.0rc6.ori/configure cdo-1.8.0rc6/configure
--- cdo-1.8.0rc6.ori/configure	2017-01-05 18:31:52.000000000 +0100
+++ cdo-1.8.0rc6/configure	2017-01-26 21:06:28.699660961 +0100
@@ -19076,6 +19076,7 @@
 
                                  NETCDF_LIBS=" -L$NETCDF_ROOT/lib -lnetcdf"
                                  NETCDF_INCLUDE=" -I$NETCDF_ROOT/include"
+export NC_CONFIG=`which nc-config`
                                  { $as_echo "$as_me:${as_lineno-$LINENO}: checking nc-config script" >&5
 $as_echo_n "checking nc-config script... " >&6; }
                                  # Extract the first word of "nc-config", so it can be a program name with args.
