diff -Nru --ignore-space-change netCDF4-1.5.4_orig/netCDF4/_netCDF4.pyx netCDF4-1.5.4/netCDF4/_netCDF4.pyx
--- netCDF4-1.5.4_orig/netCDF4/_netCDF4.pyx	2020-09-18 11:39:27.068162074 +0200
+++ netCDF4-1.5.4/netCDF4/_netCDF4.pyx	2020-09-18 11:40:28.224476927 +0200
@@ -1,3 +1,5 @@
+#cython: language_level=3
+
 """
 Version 1.5.4
 ---------------
diff -Nru --ignore-space-change netCDF4-1.5.4_orig/setup.py netCDF4-1.5.4/setup.py
--- netCDF4-1.5.4_orig/setup.py	2020-09-18 11:39:27.066122000 +0200
+++ netCDF4-1.5.4/setup.py	2020-09-18 11:49:37.330166436 +0200
@@ -175,7 +175,7 @@
 curl_incdir = os.environ.get('CURL_INCDIR')
 mpi_incdir = os.environ.get('MPI_INCDIR')
 
-USE_NCCONFIG = os.environ.get('USE_NCCONFIG')
+USE_NCCONFIG = False #os.environ.get('USE_NCCONFIG')
 if USE_NCCONFIG is not None:
     USE_NCCONFIG = bool(int(USE_NCCONFIG))
 USE_SETUPCFG = os.environ.get('USE_SETUPCFG')
diff -Nru --ignore-space-change netCDF4-1.5.4_orig/test/run_all.py netCDF4-1.5.4/test/run_all.py
--- netCDF4-1.5.4_orig/test/run_all.py	2020-09-18 11:39:27.061407000 +0200
+++ netCDF4-1.5.4/test/run_all.py	2020-09-18 11:41:09.371294000 +0200
@@ -31,12 +31,12 @@
 
 # Don't run tests that require network connectivity
 if os.getenv('NO_NET'):
-    test_files.remove('tst_dap.py');
+#    test_files.remove('tst_dap.py');
     sys.stdout.write('not running tst_dap.py ...\n')
-else:
+#else:
     # run opendap test first (issue #856).
-    test_files.remove('tst_dap.py')
-    test_files.insert(0,'tst_dap.py')
+#    test_files.remove('tst_dap.py')
+#    test_files.insert(0,'tst_dap.py')
 
 # Build the test suite from the tests found in the test files.
 testsuite = unittest.TestSuite()
