--- grib1.ORIG/src/copen_c.c	2018-11-19 14:38:12.000000000 +0100
+++ grib1/src/copen_c.c	2019-02-18 22:41:09.000000000 +0100
@@ -114,6 +114,8 @@
 #include <sys/fsid.h>
 #endif
 
+/*Setting the buffer size for the setvbuf*/
+#define LARGESIZE (1UL<<22)  // 4MiB
 /* Alle externen Variablen werden hier in copen_c deklariert, 
    deshalb wird EXTERN hier nicht, in allen anderen C Modulen
    als "extern" definiert
@@ -137,6 +139,9 @@
 
 extern int errno;
 
+/* creating an array of filebuffers of size LARGESIZE for the different streams */
+char filebuff[MAX_FILES][LARGESIZE];
+
 void copen_c ( FTN_INT *iun, char infile[], char modus[], FTN_INT *infile_len,
 	     FTN_INT *modus_len, FTN_INT *irm)
 {
@@ -449,6 +454,17 @@
     return;
  }
 
+ if (posix_fadvise(fdes[ii], 0, 0, POSIX_FADV_RANDOM) < 0) {
+     perror("posix_fadvise() failed");
+     return;
+ }
+
+/* Forcing the IO-buffer for the file streams to LARGESIZE */
+ if (setvbuf ( dateizeiger[ii], filebuff[ii], _IOFBF, LARGESIZE ) < 0) {
+    perror("setvbuf() failed");
+    return;
+ }
+ 
 /* Der Dateizeiger muss zum Anhaengen an der Stelle stehen,
    die vom Programm uegex erwartet wird.
    Zuerst Test, ob Datei bereits beschrieben, kein seek bei Erstaufruf */
