diff -Naur scorep-6.0.ori/configure scorep-6.0/configure
--- scorep-6.0.ori/configure	2019-07-31 09:06:40.064745706 +0200
+++ scorep-6.0/configure	2020-11-19 14:31:23.465258554 +0100
@@ -3144,6 +3144,8 @@
 elif test -L /opt/cray/pe/pmi/default; then :
   if test -d /opt/cray/ari/modulefiles; then :
   ac_scorep_platform="crayxc"
+elif test -d /opt/cray/pe/lmod/modulefiles/net/ofi; then :
+  ac_scorep_platform="crayxc"
 fi
 fi
                        if test "x${ac_scorep_platform}" = "x"; then :
diff -Naur scorep-6.0.ori/src/adapters/compiler/gcc-plugin/scorep_plugin_gcc_version_compatibility.h scorep-6.0/src/adapters/compiler/gcc-plugin/scorep_plugin_gcc_version_compatibility.h
--- scorep-6.0.ori/src/adapters/compiler/gcc-plugin/scorep_plugin_gcc_version_compatibility.h	2019-07-31 09:06:13.252696749 +0200
+++ scorep-6.0/src/adapters/compiler/gcc-plugin/scorep_plugin_gcc_version_compatibility.h	2020-11-19 15:06:48.959768214 +0100
@@ -43,3 +43,9 @@
 #if SCOREP_GCC_PLUGIN_TARGET_VERSION < 7000
 #define SET_DECL_ALIGN( decl, x ) DECL_ALIGN( decl ) = ( x )
 #endif
+
+#if SCOREP_GCC_PLUGIN_TARGET_VERSION < 10000
+#define cgraph_inlined_to( node ) ( node )->global.inlined_to
+#else
+#define cgraph_inlined_to( node ) ( node )->inlined_to
+#endif
diff -Naur scorep-6.0.ori/src/adapters/compiler/gcc-plugin/scorep_plugin_pass_instrument.inc.c scorep-6.0/src/adapters/compiler/gcc-plugin/scorep_plugin_pass_instrument.inc.c
--- scorep-6.0.ori/src/adapters/compiler/gcc-plugin/scorep_plugin_pass_instrument.inc.c	2019-07-31 09:06:13.240696727 +0200
+++ scorep-6.0/src/adapters/compiler/gcc-plugin/scorep_plugin_pass_instrument.inc.c	2020-11-19 15:07:52.435761282 +0100
@@ -83,7 +83,7 @@
         return "in system header";
     }
 
-    if ( node->global.inlined_to )
+    if ( cgraph_inlined_to( node ) )
     {
         return "was inlined";
     }
