diff -Naur sionlib.ori/config/determine-cc.sh sionlib/config/determine-cc.sh
--- sionlib.ori/config/determine-cc.sh	2016-11-23 14:11:31.000000000 +0100
+++ sionlib/config/determine-cc.sh	2017-08-23 13:13:59.463471000 +0200
@@ -9,33 +9,41 @@
 fi
 
 case "$CC" in
-  /opt/cray/craype/* | /opt/cray/xt*)
+  /opt/cray/craype/* | /opt/cray/pe/craype/* | /opt/cray/xt*)
       echo "$CC is probably a Cray XT compiler wrapper" ;;
   *)
       echo "Warning: $CC doesn't seem to be a Cray XT compiler wrapper!" ;;
 esac
 
+set -x
 if [ `$CC -V 2>&1 | grep -c ^Cray` -gt 0 ]; then
   COMP=cce
   COMPVER=`$CC -V 2>&1 | grep ^Cray | cut -d\  -f 5`
   echo "Got CCE ${COMPVER} compiler"
+
 elif [ `$CC -V 2>&1 | grep -c ^Intel` -gt 0 ]; then
   COMP=intel
   COMPVER=`$CC -V 2>&1 | grep ^Intel | sed -e 's/^.*Version //' | cut -d\  -f 1`
   echo "Got Intel ${COMPVER} compiler"
+
 elif [ `$CC -V 2>&1 | grep -c ^pg` -gt 0 ]; then
   COMP=pgi
   COMPVER=`$CC -V 2>&1 | grep ^pg | cut -d\  -f 2`
   echo "Got PGI ${COMPVER} compiler"
+
 elif [ `$CC -v dummy.c 2>&1 | grep -c ^PathScale` -gt 0 ]; then
   COMP=pathscale
   COMPVER=`$CC -v dummy.c 2>&1 | grep ^PathScale | sed -e 's/^.*Version //'`
   echo "Got PathScale $COMPVER compiler"
 # NB: since PathScale includes GCC, test for GCC last
+
 elif [ `$CC -v dummy.c 2>&1 | grep -c ^gcc` -gt 0 ]; then
   COMP=gnu
   COMPVER=`$CC -dumpversion`
   echo "Got GNU $COMPVER compiler"
+
 else
   echo "'$CC' compiler not identified!"
 fi
+
+set +x
