diff -Nru wannier90-1.2.orig/make.sys wannier90-1.2/make.sys
--- wannier90-1.2.orig/make.sys	2010-01-16 17:12:07.000000000 +0100
+++ wannier90-1.2/make.sys	2020-03-25 11:33:40.774367000 +0100
@@ -1,15 +1,10 @@
-
 #=================================================
-# For Linux with intel version 8/9 
+# Cray Linux with intel/19.0.1.144 
 #=================================================
-F90 = ifort
+F90 = ftn
 FCOPTS = -O2
-LDOPTS = -O2 -Vaxlib 
-#FCOPTS = -O0 -g -warn all -CB
-#LDOPTS = -O0 -Vaxlib
+LDOPTS = -O2
 #======================
 # INTEL MKL
 #======================
-LIBDIR = /home/aam/Intel/mkl721/lib/em64t
-LIBS = -L$(LIBDIR) -lmkl_lapack -lmkl_em64t /home/aam/Intel/mkl721/lib/em64t/libguide.a -lpthread
-
+LIBS = -L$(MKLROOT)/lib/intel64 -lmkl_intel_lp64 -lmkl_sequential -lmkl_core $(MKLROOT)/lib/intel64/libmkl_lapack95_lp64.a $(MKLROOT)/lib/intel64/libmkl_blas95_lp64.a
