diff -Naur likwid-4.3.2.orig/config.mk likwid-4.3.2/config.mk
--- likwid-4.3.2.orig/config.mk	2018-04-17 13:03:24.000000000 +0200
+++ likwid-4.3.2/config.mk	2018-04-19 13:49:47.203458000 +0200
@@ -10,7 +10,7 @@
 COMPILER = GCC#NO SPACE
 
 # Path were to install likwid
-PREFIX = /usr/local
+#PREFIX = /usr/local
 
 # Set the default mode for MSR access.
 # This can usually be overriden on the commandline.
@@ -20,7 +20,7 @@
 # Build Fortran90 module interface for Marker API. Adopt Fortran compiler
 # in ./make/include_<COMPILER>.mk if necessary. Default: ifort (even for
 # COMPILER=GCC)
-FORTRAN_INTERFACE = false#NO SPACE
+FORTRAN_INTERFACE = true#NO SPACE
 
 # Instrument likwid-bench with Marker API calls for use with likwid-perfctr
 INSTRUMENT_BENCH = true#NO SPACE
@@ -62,12 +62,12 @@
 BUILDDAEMON = true#NO SPACE
 # Build the setFrequencies daemon to allow users setting the CPU and Uncore
 # frequency
-BUILDFREQ = true#NO SPACE
+BUILDFREQ = false#NO SPACE
 
 # chown installed tools to this user/group
 # if you change anything here, make sure that the user/group can access
 # the MSR devices and (on Intel) the PCI devices.
-INSTALL_CHOWN = -g root -o root#NO SPACE
+#INSTALL_CHOWN = -g root -o root#NO SPACE
 
 # uncomment to optionally set external lua@5.2 or lua@5.3:
 # default is use internally provide lua
@@ -92,10 +92,10 @@
 
 # Paths to some configuration files that can be used to overwrite some
 # array lengths defined at compilation
-CFG_FILE_PATH = /etc/likwid.cfg
+CFG_FILE_PATH = $(PREFIX)/etc/likwid.cfg
 # With the likwid-genTopoCfg it is possible to store the topology of a system
 # in a file to avoid re-reading all topology informations again
-TOPO_FILE_PATH = /etc/likwid_topo.cfg
+TOPO_FILE_PATH = $(PREFIX)/etc/likwid_topo.cfg
 
 # Versioning Information
 # The libraries are named liblikwid.<VERSION>.<RELEASE>
diff -Naur likwid-4.3.2.orig/make/include_CCE.mk likwid-4.3.2/make/include_CCE.mk
--- likwid-4.3.2.orig/make/include_CCE.mk	1970-01-01 01:00:00.000000000 +0100
+++ likwid-4.3.2/make/include_CCE.mk	2018-04-19 13:57:19.508283000 +0200
@@ -0,0 +1,33 @@
+CC  = cc
+FC  = ftn
+AS  = as
+AR  = ar
+PAS = ./perl/AsmGen.pl
+GEN_PAS = ./perl/generatePas.pl
+GEN_GROUPS = ./perl/generateGroups.pl
+GEN_PMHEADER = ./perl/gen_events.pl
+
+ANSI_CFLAGS   =
+#ANSI_CFLAGS += -pedantic
+#ANSI_CFLAGS += -Wextra
+#ANSI_CFLAGS += -Wall
+
+CFLAGS   =  -O2 -hstd=c99 -hPIC
+#FCFLAGS  = -module ./  # ifort
+FCFLAGS  = -p ./  #crayftn -fsyntax-only
+PASFLAGS  = x86-64
+ASFLAGS  = 
+CPPFLAGS =
+LFLAGS   =  -pthread
+
+SHARED_CFLAGS = -hPIC # -fvisibility=hidden
+SHARED_LFLAGS = -shared # -fvisibility=hidden
+
+DEFINES  = -DPAGE_ALIGNMENT=4096
+DEFINES  += -DLIKWID_MONITOR_LOCK
+DEFINES  += -DDEBUGLEV=0
+
+INCLUDES =
+LIBS     = -lm -lrt
+
+
diff -Naur likwid-4.3.2.orig/make/include_GCC.mk likwid-4.3.2/make/include_GCC.mk
--- likwid-4.3.2.orig/make/include_GCC.mk	2018-04-17 13:03:24.000000000 +0200
+++ likwid-4.3.2/make/include_GCC.mk	2018-04-19 13:50:28.721212000 +0200
@@ -1,5 +1,5 @@
 CC  = gcc
-FC  = ifort
+FC  = gfortran
 AS  = as
 AR  = ar
 PAS = ./perl/AsmGen.pl
@@ -13,8 +13,8 @@
 #ANSI_CFLAGS += -Wall
 
 CFLAGS   =  -O2 -std=c99 -Wno-format -fPIC
-FCFLAGS  = -module ./  # ifort
-#FCFLAGS  = -J ./  -fsyntax-only  #gfortran
+#FCFLAGS  = -module ./  # ifort
+FCFLAGS  = -J ./  -fsyntax-only  #gfortran
 PASFLAGS  = x86-64
 ASFLAGS  = 
 CPPFLAGS =
