diff --git a/sklearn/setup.py b/sklearn/setup.py
index 8adbbd9..d476df2 100644
--- a/sklearn/setup.py
+++ b/sklearn/setup.py
@@ -66,7 +66,8 @@ def configuration(parent_package='', top_path=None):
     # some libs needs cblas, fortran-compiled BLAS will not be sufficient
     blas_info = get_info('blas_opt', 0)
     if (not blas_info) or (
-            ('NO_ATLAS_INFO', 1) in blas_info.get('define_macros', [])):
+            ('NO_ATLAS_INFO', 1) in blas_info.get('define_macros', [])) or (
+            ('HAVE_CBLAS', None) in blas_info.get('define_macros', [])):
         config.add_library('cblas',
                            sources=[join('src', 'cblas', '*.c')])
         warnings.warn(BlasNotFoundError.__doc__)
