diff -Nru gpaw-1.2.0.orig/config.py gpaw-1.2.0/config.py
--- gpaw-1.2.0.orig/config.py	2017-01-06 10:52:59.000000000 +0100
+++ gpaw-1.2.0/config.py	2017-02-10 14:44:27.394724247 +0100
@@ -141,6 +141,14 @@
             if openblas:  # prefer openblas
                 libraries += ['openblas']
                 library_dirs += [libdir]
+# Cray systems: check if cray-libsci is available
+            craylibsci = False
+            if 'CRAY_LIBSCI_DIR' in os.environ:
+                craylibsci = True
+                os.environ["GPAW_MPI"] = "True"
+                os.environ["GPAW_MPI_IMPLEMENTATION"] = "cray"
+                os.environ["GPAW_MPI_COMMAND"] = "srun"
+                libraries += ['readline', 'ncurses']
             else:
                 if atlas:  # then atlas
                     # http://math-atlas.sourceforge.net/errata.html#LINK
@@ -347,6 +355,9 @@
     elif mpi == 'poe':
         mpicompiler = 'mpcc_r'
 
+    elif mpi == 'cray':
+        mpicompiler = 'cc'
+
     else:
         # Try to use mpicc
         mpicompiler = 'mpicc'
diff -Nru gpaw-1.2.0.orig/customize.py gpaw-1.2.0/customize.py
--- gpaw-1.2.0.orig/customize.py	2017-01-06 10:52:59.000000000 +0100
+++ gpaw-1.2.0/customize.py	2017-02-10 14:38:40.689074683 +0100
@@ -27,7 +27,7 @@
 
     libraries += ['somelib', 'otherlib']
 """
-
+import os
 # compiler = 'gcc'
 # mpicompiler = 'mpicc'  # use None if you don't want to build a gpaw-python
 # mpilinker = 'mpicc'
@@ -76,7 +76,14 @@
 
 
 # Build MPI-interface into _gpaw.so:
-if 0:
-    compiler = 'mpicc'
-    define_macros += [('PARALLEL', '1')]
-    mpicompiler = None
+if os.environ["GPAW_MPI"] == "True" and os.environ["GPAW_MPI_IMPLEMENTATION"] == "cray":
+   print ("Cray Libraries found on the system: building with cray wrappers, cray-libsci and cray-mpich")
+   os.environ["CRAYPE_LINK_TYPE"] = "dynamic"
+   compiler = 'cc'
+   mpicompiler = 'cc'
+   mpilinker = 'cc'
+   define_macros += [('PARALLEL', '1')]
+else:
+   compiler = 'mpicc'
+   define_macros += [('PARALLEL', '1')]
+   mpicompiler = None
