diff --git deepmd-kit-1.2.1.orig/source/lib/include/common.h deepmd-kit-1.2.1/source/lib/include/common.h
index 3912f21f..620c88eb 100644
--- deepmd-kit-1.2.1.orig/source/lib/include/common.h
+++ deepmd-kit-1.2.1/source/lib/include/common.h
@@ -1,14 +1,22 @@
 #pragma once
 
 #include "tensorflow/core/public/session.h"
+#include "tensorflow/core/public/version.h"
 #include "tensorflow/core/platform/env.h"
 #include "tensorflow/core/framework/op.h"
 #include "tensorflow/core/framework/op_kernel.h"
 #include "tensorflow/core/framework/shape_inference.h"
+#include <string>
 
 using namespace tensorflow;
 using namespace std;
 
+#if TF_MAJOR_VERSION >= 2 && TF_MINOR_VERSION >= 2
+typedef tensorflow::tstring STRINGTYPE;
+#else
+typedef std::string STRINGTYPE;
+#endif
+
 #include "NNPAtomMap.h"
 #include <vector>
 #include "version.h"
diff --git deepmd-kit-1.2.1.orig/source/lib/src/DataModifier.cc deepmd-kit-1.2.1/source/lib/src/DataModifier.cc
index 65b3f85d..03fcd43a 100644
--- deepmd-kit-1.2.1.orig/source/lib/src/DataModifier.cc
+++ deepmd-kit-1.2.1/source/lib/src/DataModifier.cc
@@ -37,7 +37,7 @@ init (const string & model,
   rcut = get_scalar<VALUETYPE>("descrpt_attr/rcut");
   cell_size = rcut;
   ntypes = get_scalar<int>("descrpt_attr/ntypes");
-  model_type = get_scalar<string>("model_attr/model_type");
+  model_type = get_scalar<STRINGTYPE>("model_attr/model_type");
   get_vector<int>(sel_type, "model_attr/sel_type");
   sort(sel_type.begin(), sel_type.end());
   inited = true;
diff --git deepmd-kit-1.2.1.orig/source/lib/src/DeepTensor.cc deepmd-kit-1.2.1/source/lib/src/DeepTensor.cc
index bd3a0ea1..9b9715fd 100644
--- deepmd-kit-1.2.1.orig/source/lib/src/DeepTensor.cc
+++ deepmd-kit-1.2.1/source/lib/src/DeepTensor.cc
@@ -33,7 +33,7 @@ init (const string & model,
   rcut = get_scalar<VALUETYPE>("descrpt_attr/rcut");
   cell_size = rcut;
   ntypes = get_scalar<int>("descrpt_attr/ntypes");
-  model_type = get_scalar<string>("model_attr/model_type");
+  model_type = get_scalar<STRINGTYPE>("model_attr/model_type");
   odim = get_scalar<int>("model_attr/output_dim");
   get_vector<int>(sel_type, "model_attr/sel_type");
   inited = true;
diff --git deepmd-kit-1.2.1.orig/source/lib/src/NNPInter.cc deepmd-kit-1.2.1/source/lib/src/NNPInter.cc
index aea9d48c..c1347504 100644
--- deepmd-kit-1.2.1.orig/source/lib/src/NNPInter.cc
+++ deepmd-kit-1.2.1/source/lib/src/NNPInter.cc
@@ -639,7 +639,7 @@ compute (ENERGYTYPE &			dener,
 void
 NNPInter::
 get_type_map(std::string & type_map){
-    type_map = get_scalar<std::string>("model_attr/tmap");
+    type_map = get_scalar<STRINGTYPE>("model_attr/tmap");
 }
 
 
